distributed/worker.py

```diff
@@ -1199,22 +1266,9 @@ async def heartbeat(self):
         finally:
             self.heartbeat_active = False
 
+    @fail_hard
     async def handle_scheduler(self, comm):
-        try:
-            await self.handle_stream(comm, every_cycle=[self.ensure_communicating])
-        except Exception as e:
-            logger.exception(e)
-            raise
-        finally:
-            if self.reconnect and self.status in Status.ANY_RUNNING:
-                logger.info("Connection to scheduler broken.  Reconnecting...")
-                self.loop.add_callback(self.heartbeat)
-            else:
-                logger.info(
-                    "Connection to scheduler broken. Closing without reporting.  Status: %s",
-                    self.status,
-                )
-                await self.close(report=False)
+        await self.handle_stream(comm, every_cycle=[self.ensure_communicating])
```

Comment:
This is huge. I'm not against it but think this would warrant a dedicated PR even if it's just about changelog documentation and/or bisecting

Reply:
there are still remanants of reconnect around. If we actually want to remove this feature we should purge everything, i.e. remove Worker.reconnect (or deprecate) and remove https://github.com/dask/distributed/blob/b8370030994b36c0625763f3d7c970e1ad059cae/distributed/worker.py#L1171-L1179

Reply:
what _is_ the behavior now if the connection severs?